### ISSUE TITLE
`db-migrate db:create` not working as intended

### ISSUE THREAD

**Issue by StlthyLee** [db-migrate/node-db-migrate#468]
I've been playing around for most of the morning with db-migrate and getting myself familiar with the ins and outs. One thing I have noticed is that the db:create command doesn't seem to work as I would have expected it. 

For example in my `database.json` I have configuration for a test database and a dev database. Now, when I run the `db-migrate db:create myapptest` I would expect to see that it creates a database to run my tests against. instead I get the following error: 

`[ERROR] Error: ER_BAD_DB_ERROR: Unknown database 'myapptest'` 

the whole reason for running the `db:create` command is _because_ that database doesn't exist and I want to create it.

Yet if I create the database and then run the command passing a a random name it will happily create any database I want. It will also claim that I have created the same database multiple times even when it has in fact created it once.

**Comment by c100k** [comment]
Just has the same problem. Maybe you should rename the issue because that is a big issue.
`db:create tries to connect to db before creating it`.

**Comment by deegale** [comment]
I ran into this as well when trying to set up a new project using db-migrate. 

If you remove the database name from `database.json`, then `db-migrate db:create mydatabase` works fine. But then none of the migrations will work until you put the database name back in `database.json`

What I expected was:

- running `db-migrate db:create` (no database name specified) will attempt to create a database with the name as specified in `database.json` If there is nothing specified, it will error out.
- running `db-migrate db:create nameofdb` will attempt to create "nameofdb", if it doesn't already exist. If it does already exist, it will report that. This will override whatever database is specified in `database.json`

**Comment by bernardocs** [comment]
I'm also having the same issue.

If I don't specify the database at `database.json` and run `db:create [db name]` it will create successfully. But if any database is specified at the configuration it'll try to connect to it before running anything.

I think the expected behavior should be like @deegale described.

**Comment by zaidqureshi2** [comment]
I am having an issue where when I do `db-migrate db:create <dbname>`  it errors out saying database "zaidq" does not exist where I think "zaidq" is my pc user

**Comment by oroce** [comment]
@zaidqureshi2 @bernardocs @deegale I just solved with some magic:

TLDR: Turning off the magic flag of `CONNECT_WITH_DB`: https://github.com/mysqljs/mysql#default-flags

First I have the following `.env`:

```
MYSQL_URL=mysql://root:root@localhost/myapp
```

and the database.json:

```
{
  "defaultEnv": "mysql",
  "mysql": {
      "driver": "mysql",
      "url": { "ENV": "MYSQL_URL" },
      "multipleStatements": true
   }
}
```

So a basic configuration. The `db-migrate db:create myapp` fails because the database does not exists. So I either remove the database from the url but then my application won't know about which db should be used or I turn off the flag of `CONNECT_WITH_DB` in the mysql connection, but I want to do this only for the db:create, for the rest of the commands I want to have database there. So this is what I did:
```
{
  "defaultEnv": "mysql",
  "mysql": {
      "driver": "mysql",
      "url": { "ENV": "MYSQL_URL" },
      "flags": { "ENV": "MYSQL_FLAGS" },
      "multipleStatements": true
   }
}
```

The commands I use:

* db creation: `MYSQL_FLAGS="-CONNECT_WITH_DB" db-migrate sync`
* db sync: `db-migrate sync`

Since the creation and syncing are two separate steps a new connection will be created everytime.

Hope this helps you!

**Comment by aprakash-sovrn** [comment]
This occurs when trying to use Postgres as well. My driver is "pg", and I've installed `db-migrate-pg`.

```
>  db-migrate db:create $( jq ".local.database" database.json) -e local[ERROR] AssertionError [ERR_ASSERTION]: ifError got unwanted exception: database "aegisdb" does not exist
    at /usr/local/lib/node_modules/db-migrate/lib/commands/db.js:17:12
    at /usr/local/lib/node_modules/db-migrate/lib/driver/index.js:95:9
    at /usr/local/lib/node_modules/db-migrate-pg/index.js:662:7
```

On db-migrate 0.11.3, btw.

**Comment by RenWenshan** [comment]
Having the same issue, I guess it tries to connect first using the config, then run db:create

**Comment by RenWenshan** [comment]
I've added a fix, ready for review @wzrdtales

**Comment by woodske** [comment]
This has not been resolved in `0.11.14`

**Comment by mckayreedmoore** [comment]
Running 0.11.14 as well. Workaround still works but issue still exists.
